iree_tests/conftest.py

```diff
@@ -206,45 +212,54 @@ def discover_test_cases(self):
         skip_missing = self.config.getoption("skip_tests_missing_files")
         have_lfs_files = self.check_for_lfs_files()
 
-        test_data_flagfile_name = "test_data_flags.txt"
-        if (self.path.parent / test_data_flagfile_name).exists():
+        mlir_files = sorted(self.path.parent.glob("*.mlir*"))
+        if len(mlir_files) == 0:
+            if not skip_missing:
+                raise FileNotFoundError(
+                    f"Missing .mlir file for test {self.path.parent.name}"
+                )
+            return test_cases
+        assert len(mlir_files) <= 1, "Test directories may only contain one .mlir file"
+        mlir_file = mlir_files[0]
+
+        if self.path.name == TEST_DATA_FLAGFILE_NAME:
             test_cases.append(
-                MlirFile.TestCase(
-                    name="test",
-                    runtime_flagfile=test_data_flagfile_name,
+                MlirCompileRunTest.TestCase(
+                    name="",
+                    mlir_file=mlir_file,
+                    runtime_flagfile=TEST_DATA_FLAGFILE_NAME,
                     enabled=have_lfs_files,
                 )
             )
```

Comment:
Could remove the need for this branch by generating boilerplate `test_cases.json` files in the ONNX test suite:

```json
{
  "file_format": "test_cases_v0",
  "test_cases": [
    {
      "name": "",
      "runtime_flagfile": "test_data_flags.txt",
      "remote_files": []
    }
  ]
}
```

Trying to keep this conftest file _somewhat_ simple and a bit concerned this is sliding the wrong way 🤔

Reply:
Yeah, would probably be best to have test_cases.json files for all the onnx tests too. That way all our tests can follow a unified format and the conftest won't get confusing. We should probably be anchoring on a test configuration file rather than the runtime flag file for onnx tests

Reply:
Okay, I can make that change too. How does a follow-up PR sound? That will require touching lots of files.

Reply:
Might also be able to replace `test_cases.json` with `tests.py` and then further simplify (or remove) the `conftest.py` file... 🤔 Mainly trying to keep the large test suites autogenerated, concise, and directly compatible with `iree-compile` -> `iree-run-module`.